destreamer.ts

```diff
@@ -264,18 +269,35 @@ async function getVideoInfo(videoID: string, session: any) {
 function getVideoUrls() {
     const t = argv.videoUrls[0] as string;
     const isPath = t.substring(t.length-4) === '.txt';
+    const regex = /(?:https:\/\/)?web.microsoftstream.com\/video\/[a-z0-9]{8}-(?:[a-z0-9]{4}\-){3}[a-z0-9]{12}/;

```

Comment:
This is not going to match private tenants (tenantName.microsoftstream.com) and custom domains (videos.mycompany.com). We should be much more relaxed on the validation rules, i'm not convinced we need to use regex. I'm open to suggestions.

Also i would suggest moving all the URL validation to its own function. Once validated any logic that consumes URLs will no longer need to do any further checks, just take it for granted.

Reply:
oh, didn't know that
maybe we can check the /video/guid part only?
if it has no match either, we can probably remove this.

ok, i'll create a new function

Reply:
That's a safer check, i agree, we should also allow for query parameters since users may copy paste things from the browser's address bar and that usually has `?list=trending` trailing the URL.

Reply:
`(?:https:\/\/)?.*\/video\/[a-z0-9]{8}-(?:[a-z0-9]{4}\-){3}[a-z0-9]{12}\??(?:&?[^=&]*\=?[^=&]*)*`

better?

Reply:
Better. Let's use this for now.